presto-main/src/main/java/com/facebook/presto/execution/SqlStageExecution.java

```diff
@@ -520,6 +586,13 @@ else if (taskState == TaskState.FINISHED) {
         }
     }
 
+    private boolean isRecoverable(List<ExecutionFailureInfo> failures)
+    {
+        return stageTaskRecoveryCallback.isPresent() &&
+                failedTasks.size() + 1 < allTasks.size() * maxFailedTaskPercentage &&
+                failures.stream().map(ExecutionFailureInfo::getErrorCode).allMatch(RECOVERABLE_ERROR_CODES::contains);
```

Comment:
I usually just writes an for-loop for such case. Use stream doesn't seem to have readability benefit here , and imperative programming style is also quite self-explained.